backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/impl/LuceneNumericFieldSortBuilder.java

```diff
@@ -39,17 +39,15 @@ public void buildAndContribute(LuceneSearchSortCollector collector) {
 
 	private SortField createSortField() {
 		LuceneNumericDomain<E> numericDomain = codec.getDomain();
+
+		LuceneNumericFieldComparatorSource fieldComparatorSource = new LuceneNumericFieldComparatorSource(
+				nestedDocumentPath, numericDomain, (E) getEffectiveMissingValue( missingValue, order ) );
+
 		if ( nestedDocumentPath != null ) {
-			nestedFieldSort = new NestedNumericFieldComparatorSource( nestedDocumentPath, numericDomain, (E) getEffectiveMissingValue( missingValue, order ) );
-			return new SortField( absoluteFieldPath, nestedFieldSort, order == SortOrder.DESC );
+			// propagate the comparator source, so that eventually it can receive the original query,
+			// that is required to nested document field on the fly sorting
+			this.nestedFieldSort = fieldComparatorSource;
```

Comment:
I just remembered that `LuceneNumericFieldSortBuilder.buildAndContribute()` can be called concurrently from multiple threads. The mutating methods cannot, but `buildAndContribute` can. We could change that, but that would require adding yet another set of interfaces and classes, and, well... Not today.

Since `fieldComparatorSource` is mutable, this assignment could cause trouble.

Maybe you could just move this code to `buildAndContribute` and make `nestedFieldSort` a local variable?

Reply:
You're right. There is no reason to extract a field here.